conjur/credentials_from_file.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+
+"""
+CredentialsFromFile module
+
+This module holds the logic for writing user-specific credentials
+to a netrc file on the user's machine when they login
+"""
+
+# Builtins
+import logging
+import netrc
+import os
+import stat
+
+# Internals
+from conjur.constants import DEFAULT_NETRC_FILE
+
+class CredentialsFromFile:
+    """
+    CredentialsFromFile
+
+    This class holds logic when credentials are kept in the netrc
+    """
+    @classmethod
+    def save(cls, credential_data):
+        """
+        Method that writes user data to a netrc file
+        and updates permissions on the file
+        """
+        with open(DEFAULT_NETRC_FILE, 'w') as netrc_file:
+            # pylint: disable=line-too-long
+            netrc_file.write(f"machine {credential_data.machine}\nlogin {credential_data.login}\npassword {credential_data.password}\n")
+            netrc_file.close()
+
+        # TODO validate this isn't a problem on Windows
+        # TODO validate _netrc on Windows
+        # Ensures that the netrc file is only available to owner
+        os.chmod(DEFAULT_NETRC_FILE, stat.S_IRWXU)
```

Comment:
If I am understanding correctly, moving `api_key`, `login_id` to a constructor of the class and the return of load will still be a dictionary?
```
def __init__(api_key, login_id):
  self.api_key=api_key
...
def load()
  loaded_netrc[self.api_key]= api_key
```
?

Reply:
>  instead of "credential_data" we will have a strong typed data that we get from the user in init

Not sure what you mean here because load data doesn't get filled from the user, but from the actual file (netrc)

Reply:
as discussed this change will touch the Client.py which at this time I am don't want to touch because of backwards compatibility